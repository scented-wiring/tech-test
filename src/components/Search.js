import React, { useState } from "react";
import "../styles/Search.css";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();
    setSearchResults(getImages(value));
  };

  return (
    <>
      <form className="Form" onSubmit={handleSubmit}>
        <input
          type="text"
          className="Input"
          onChange={(e) => setValue(e.target.value)}
        ></input>
        <button className="Button" type="submit">
          Search
        </button>
      </form>
    </>
  );
};

export default Search;
