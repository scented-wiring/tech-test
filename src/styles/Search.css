.Form {
  margin: auto;
  width: 30%;
  display: flex;
  flex-direction: row;
}

.Input {
  display: block;
  width: 100%;
}

.Button {
  color: #fff !important;
  background: red;
  padding: 8px;
  border: solid 1px black;
  transition: all 0.4s ease 0s;
  font-weight: bold;
}

.Button:hover {
  color: red !important;
  background: white;
}
