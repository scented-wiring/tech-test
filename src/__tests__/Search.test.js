import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const { asFragment } = render(<Search />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a text input form", () => {
    render(<Search />);
    const input = screen.getByTestId("input");

    expect(input).toHaveAttribute("type", "text");
  });

  it("renders a submit button", () => {
    render(<Search />);
    const button = screen.getByTestId("button");

    expect(button).toHaveAttribute("type", "submit");
  });
});
