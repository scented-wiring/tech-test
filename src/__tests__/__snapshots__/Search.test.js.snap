// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="Form"
  >
    <input
      class="Input"
      type="text"
    />
    <button
      class="Button"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
